modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/encoding/target/TargetEncoderPreprocessor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.preprocessing.encoding.target;
+
+import java.util.Set;
+import org.apache.ignite.ml.math.primitives.vector.VectorUtils;
+import org.apache.ignite.ml.preprocessing.Preprocessor;
+import org.apache.ignite.ml.preprocessing.encoding.EncoderPreprocessor;
+import org.apache.ignite.ml.structures.LabeledVector;
+
+/**
+ * Preprocessing function that makes Target encoding.
+ *
+ * The Target Encoder Preprocessor encodes string values (categories) to double values
+ * in range [0.0, 1], where the value will be presented as a regularized mean target value of
+ * a category.
+ *
+ * alpha = 1 / (1 + exp(-(categorySize - min_samples_leaf) / beta))
+ * encodedValue = globalTargetMean * (1 - alpha) + categoryTargetMean * alpha
+ * if categorySize == 1 then use globalTargetMean
+ *
+ * min_samples_leaf - minimum samples to take category average into account.
```

Comment:
looks like min_samples_leaf is not used in this class

Reply:
Yes, but it is used for evaluate `TargetEncodingMeta`, so I wanted to mention this in encoder class.